### PR TITLE
Add Raspberry Pi

### PR DESCRIPTION
**Pull request by mudge**
Add the user account system for parts of https://www.raspberrypi.com ([#16,036 in Similarweb](https://www.similarweb.com/website/raspberrypi.com/#overview)), e.g. https://events.raspberrypi.com

### CODE REVIEW

**Comment by mudge** [comment]
It looks like the build is failing due to [Similarweb returning a `429 Too Many Requests`](https://app.circleci.com/pipelines/github/2factorauth/twofactorauth/8379/workflows/d579aecf-1f58-4c52-9dbd-74b06ec30739/jobs/8403) but we also have the issue that this is an unranked subdomain (`id.raspberrypi.com`) of a ranked domain (`www.raspberrypi.com`).

**Comment by Carlgo11** [comment]
Thanks for the pull request @mudge!
There are a few issues with your PR. Firstly, just as we don’t list account.google.com separately from Google and we don’t list appleid.apple.com, the login portal for raspberrypi.com isn’t something we’re going to list.
If you instead would like to list just raspberrypi.com, perhaps under the retail category, that’s something which we’d be able to accept.

**Comment by mudge** [comment]
Thanks for the prompt response, @Carlgo11.

To clarify, is the issue here the name and category of the entry rather than the domain as I see [`appleid.apple.com`](https://github.com/2factorauth/twofactorauth/blob/master/entries/a/appleid.apple.com.json) and [`id.atlassian.com`](https://github.com/2factorauth/twofactorauth/blob/master/entries/i/id.atlassian.com.json) as entries using the most specific subdomain possible? I'm keen to avoid confusion with other parts of raspberrypi.com which don't have 2FA, e.g. https://forums.raspberrypi.com

In any case, I've updated the PR to use "Raspberry Pi" as the name and "retail" as the category.

**Comment by Carlgo11** [comment]
To clarify,
the "Identity Management" category is meant for sites that primarily offer SAML/LDAP/SSO services to their customers. An account management page only intended to integrate with a single company doesn't qualify for that category.
An entry also can't link directly to a login page. Instead, it should link to a service using the login page.
Due to this, `id.raspberrypi.com` can't be listed by itself on 2FA Directory but the services using Raspberry Pi ID can. I don't really know which services Raspberry Pi ID integrates with. Maybe you can shed some light on that?

**Comment by mudge** [comment]
The plan is for it to be used across various services across https://www.raspberrypi.com (effectively a single sign-on across Raspberry Pi Ltd sites which is why it only loads a log in page if you try to go to it directly).

At the moment, it is only used to log in to [Raspberry Pi Events](https://events.raspberrypi.com) for the submission and management of community events but we also plan to use it for our [Product Information Portal](https://pip.raspberrypi.com) where users can download compliance documents (some requiring an NDA) for various Raspberry Pi products.

Perhaps I am contributing this too early? I was keen for its inclusion in your API as a prompt to password managers that any account registered on https://id.raspberrypi.com (as that is where the sign in and sign up page will always live, despite it being used on sites on different subdomains of raspberrypi.com) has two-factor authentication available.

**Comment by hkamran80** [comment]
Is there a reason why `id.raspberrypi.com` has to be listed? Why can't it just be `raspberrypi.com`?

**Comment by mudge** [comment]
@hkamran80 now updated so the entry is for all of `raspberrypi.com` rather than `id.raspberrypi.com`.

I initially wanted to avoid confusion as `forums.raspberrypi.com` doesn't support 2FA and was thinking of password managers consuming your API falsely reporting that accounts there could enable it but that's a specific edge case we can work on.

**Comment by soodaayush** [comment]
This PR has been open for a month now and it seems it can be merged now.

File: img/r/raspberrypi.com.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...4 9.98-9.79 9.98s-16.82-4.137-9.79-9.98a15.63 15.63 0 0 1 19.58 0ZM39.028 57.145a15.743 15.743 0 0 1 10.246-8.599c5.832-1.057 8.239.903 4.536 5.302a35.232 35.232 0 0 1-10.324 8.654c-4.915 2.806-6.53.156-4.458-5.357ZM28.86 87.776a15.738 15.738 0 0 1 1.168-13.392c2.598-4.419 5.212-4.243 6.249-.5a23.48 23.48 0 0 1-1.353 15.476c-1.923 3.823-4.329 2.531-6.064-1.584ZM47.588 116.4a15.645 15.645 0 0 1-11.768-13.963c.975-22.688 33.673 17.829 11.768 13.963ZM48.254 88.143c-5.483-3.166-6.934-10.918-3.24-17.315s11.131-9.017 16.615-5.851c5.483 3.165 6.934 10.918 3.24 17.315s-11.132 9.016-16.615 5.85ZM80.016 125.058a15.658 15.658 0 0 1-18.3 0c-5.433-3.752.033-8.069 9.15-8.069s14.583 4.317 9.15 8.07Z" class="cls-2"/><ellipse cx="70.866" cy="100.396" class="cls-2" rx="12.728" ry="10.909"/><path d="M76.863 82.292c-3.694-6.397-2.243-14.15 3.24-17.315 5.484-3.166 12.923-.547 16.616 5.85s2.242 14.15-3.24 17.316c-5.484 3.165-12.923.546-16.616-5.851ZM94.144 116.4c-21.904 3.866 10.793-36.651 11.768-13.963A15.645 15.645 0 0 1 94.144 116.4ZM112.872 87.776c-1.735 4.115-4.14 5.407-6.064 1.584a23.48 23.48 0 0 1-1.353-15.475c1.038-3.744 3.652-3.92 6.249.5a15.738 15.738 0 0 1 1.168 13.391Z" class="cls-2"/></svg>
```
Comment (hkamran80) [inline img/r/raspberrypi.com.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 99 128"><path d="M99 73c0-6-3-11-7-14a17 17 0 0 0-3-6 19 19 0 0 0-7-13 7 7 0 0 0 2-2c4-2 7-6 8-9a13 13 0 0 0 4-9 8 8 0 0 0-1-3 7 7 0 0 0 2-8c-1-3-6-6-13-6a10 10 0 0 0-7-3 16 16 0 0 0-3 1 9 9 0 0 0-4-1 15 15 0 0 0-7 2 7 7 0 0 0-1 0c-3 0-7 2-8 3a24 24 0 0 0-4 5 24 24 0 0 0-5-5c-1-1-4-3-8-3a15 15 0 0 0-8-2 9 9 0 0 0-3 1 16 16 0 0 0-4-1 10 10 0 0 0-7 3C9 3 4 6 2 9a7 7 0 0 0 2 8 8 8 0 0 0 0 3 13 13 0 0 0 3 9c1 3 4 7 9 9a7 7 0 0 0 1 2 19 19 0 0 0-7 13 17 17 0 0 0-3 6c-4 3-6 8-7 14 0 6 1 12 4 15a21 21 0 0 0 4 8 21 21 0 0 0 5 13 22 22 0 0 0 12 7 29 29 0 0 0 10 6 21 21 0 0 0 15 6 21 21 0 0 0 15-6 29 29 0 0 0 9-6 22 22 0 0 0 12-7 21 21 0 0 0 5-13 21 21 0 0 0 4-8c3-3 5-9 4-15z"/><path fill="#46af4b" d="M62 9c0 2 1 1 1 1 3-3 5-3 7-3a1 1 0 0 1 1 1v1c4-2 7-2 9-1-2 1-1 2 0 2 3-1 9 0 10 1a1 1 0 0 1 0 1 10 10 0 0 0-4 7h2a1 1 0 0 1 1 1c0 2-2 4-5 6l1 1a1 1 0 0 1 0 1c-1 2-2 4-7 5a1 1 0 0 0 0 1v1c-5 4-18 3-19-5a48 48 0 0 1 20-16v-1a34 34 0 0 0-22 14 1 1 0 0 1-1 0c-8-4-2-16 6-18zM14 28a1 1 0 0 1 1-1v-1c-3-2-5-4-5-6a1 1 0 0 1 1-1h2a10 10 0 0 0-4-7 1 1 0 0 1 0-1c2-1 7-2 11-1 0 0 1-1-1-2a1 1 0 0 1 1 0c1-1 4-1 8 1V8a1 1 0 0 1 1-1c2 0 5 0 7 3 0 0 1 1 1-1a1 1 0 0 1 1 0c7 2 13 14 5 18a34 34 0 0 0-23-14v1a48 48 0 0 1 21 16c-2 8-14 9-19 5v-1a1 1 0 0 0 0-1c-5-1-7-3-8-5z"/><path fill="#cd2355" d="M81 50c3 6 1 8-4 5a35 35 0 0 1-10-8c-4-5-2-7 4-6a16 16 0 0 1 10 9zm-22-6c7 6-1 10-9 10-9 0-17-4-10-10a16 16 0 0 1 19 0zm-41 6a16 16 0 0 1 10-9c6-1 8 1 5 6a35 35 0 0 1-11 8c-5 3-6 1-4-5zM8 81a16 16 0 0 1 1-14c2-4 5-4 6 0a23 23 0 0 1-1 15c-2 4-5 3-6-1zm18 28a16 16 0 0 1-11-14c1-22 33 18 11 14zm1-28c-5-3-7-11-3-17 3-7 11-9 16-6 6 3 7 11 4 17-4 7-11 9-17 6zm32 37a16 16 0 0 1-19 0c-5-4 1-8 10-8s14 4 9 8z"/><ellipse cx="49.6" cy="93.3" fill="#cd2355" rx="12.7" ry="10.9"/><path fill="#cd2355" d="M56 75c-4-6-3-14 3-17 5-3 13-1 16 6 4 6 3 14-3 17-5 3-13 1-16-6zm17 34c-22 4 11-36 12-14a16 16 0 0 1-12 14Zm19-28c-2 4-4 5-6 1a23 23 0 0 1-2-15c1-4 4-4 6 0a16 16 0 0 1 2 14z"/></svg>
```

**Review by hkamran80 — CHANGES_REQUESTED** [review]

File: img/r/raspberrypi.com.svg (hunk 1 above)
Comment (mudge) [reply img/r/raspberrypi.com.svg]: Updated in a51c061fe48646e14d972eff731572e559fffffb.

**Review by hkamran80 — APPROVED** [review]

**Comment by hkamran80** [comment]
Thank you! Your contribution will be visible in around 10 minutes, although it may take longer due to server and browser caches.

**Pull request merged**